src/crypto/openssl/key_pair.cpp

```diff
@@ -281,35 +315,21 @@ namespace crypto
     OpenSSL::CHECK1(X509_add_ext(crt, ext, -1));
     X509_EXTENSION_free(ext);
 
-    // Subject alternative names (Necessary? Shouldn't they be in the CSR?)
-    if (!subject_alt_names.empty())
+    // Add subject alternative names (read from csr)
+    Unique_STACK_OF_X509_EXTENSIONS exts = X509_REQ_get_extensions(csr);
+    int extension_count = sk_X509_EXTENSION_num(exts);
+    if (extension_count > 0)
     {
-      std::string all_alt_names;
-      bool first = true;
-      for (auto san : subject_alt_names)
+      for (size_t i = 0; i < extension_count; i++)
       {
-        if (first)
+        X509_EXTENSION* ext = sk_X509_EXTENSION_value(exts, i);
+        ASN1_OBJECT* obj = X509_EXTENSION_get_object(ext);
+        auto nid = OBJ_obj2nid(obj);
+        if (nid == NID_subject_alt_name)
```

Comment:
Here, we only copy the SANs extension from the CSR to the final certificate. I don't think there's a use case for it yet, but should we be more permissive and copy _all_ extensions from the CSR?

Reply:
Signing things we haven't validated sounds dangerous?

Reply:
+1, I also think we should just keep the SANs.